src/Microsoft.ComponentDetection.Contracts/TypedComponent/CargoComponent.cs

```diff
@@ -9,20 +9,38 @@ private CargoComponent()
             // reserved for deserialization
         }
 
-        public CargoComponent(string name, string version)
+        public CargoComponent(string name, string version, string checksum = null, string source = null)
         {
             Name = ValidateRequiredInput(name, nameof(Name), nameof(ComponentType.Cargo));
             Version = ValidateRequiredInput(version, nameof(Version), nameof(ComponentType.Cargo));
+            Checksum = checksum;
+            Source = source;
         }
 
         public string Name { get; set; }
 
         public string Version { get; set; }
 
+        public string Checksum { get; set; }
+
+        public string Source { get; set; }
+
         public override ComponentType Type => ComponentType.Cargo;
 
-        public override string Id => $"{Name} {Version} - {Type}";
+        public override string Id
+        {
+            get {
+                if (Source != null)
+                {
+                    return $"{Name} {Version} {Source} - {Type}";
```

Comment:
If it does make sense to keep `Source` in the `Id` string, we may want to put it in parentheses `({Source})`, which better matches what's in the lock file.

Reply:
`Id` is used for identity, so if it is not included in this string, then matching nodes will be merged as part of the detection.

I have added the parentheses as suggested. Still waiting to validate if/how this might cause issues for other tools downstream.